### ISSUE TITLE
Handle training app cut over to production app better

### ISSUE THREAD

**Issue by mrjones-plip** [medic/cht-core#6668]
**Is your feature request related to a problem? Please describe.**
CHWs are often trained on one app which uses a test cht-core instance and then expected to use another app for live data entry on the production cht-core instance.  This can result in confusion as to which app to use, [most recently 50 CHWs entered 1800 documents in a test instance](https://github.com/medic/config-moh-mali/issues/98) that all should have been in the live instance.

**Describe the solution you'd like**
cht-core should offer a way to explicitly allow for CHW training in a way that both allows using the app that's as close to production as possible while still offering an explicit, and not confusing way to stop using the test instance on the go live data. This might include, but should not be limited to:
* a "test server" setting such that the app and web UI are very clearly branded "THIS IS A TEST INSTANCE", possibly adding a "DO NOT USE AFTER {{SPECIFIC_DATE}}" 
* Offer a routing option that before the go live date, all traffic is pushed to the test server.  After the go live data, with out any changes on the CHW's device, traffic is transparently pushed to the production server (this presents some issues on sessions and users persisting between the two back ends of course)
* Having the test app cease to function after the go live date, even if the CHW is offline
* Have automatic reminders in the test app to not use it after a certain date
* Allow a test app to push it's data to a different server when it's been discovered it's been used to gather live data.
* Alert administrators when CHWs are uploading docs to a test instance after a certain date

**Describe alternatives you've considered**
Some solutions we've thought about:
* change all passwords for CHWs on a specific go live date
* installing the [generic app for testing](https://play.google.com/store/apps/details?id=org.medicmobile.webapp.mobile&hl=en_US), then telling CHW to use [the real app](https://play.google.com/store/apps/details?id=org.medicmobile.webapp.mobile.covid_moh_mali) on a certain date
* manually migrating data, after it's been synced, from the test server -> production server

**Additional context**
A big problem we've faced is accounting for the offline first nature of cht-core.  That is, if a CHW is one of 50 CHWs entering data into the wrong system, preemptively changing their password _before_ they sync up to the server, means data is lost.  Data loss should be avoided at all costs.

FYI - @kennsippell , @binokaryg , @Marina-Kenf , @abbyad & @garethbowen

**Comment by MaxDiz** [comment]
Added to 3.13 since this is an ongoing challenge for deployment management. It would be good to do another review with the App Services team to make sure the solutions in this ticket are globally applicable to deployment processes.

**Comment by mrjones-plip** [comment]
yeah 0 thanks @joyannee ! I think when we gather requirements for this from the field and pair that against level of effort from Core Eng, we'll come up with some neat solutions.
